scripts/test-helm.sh

```diff
@@ -152,12 +153,12 @@ then
   exit 1
 fi
 echo "ok."
-echo -n "test-helm.sh] uninstalling the Helm Chart $HELM_CHART_NAME in namespace $K8S_NAMESPACE ... "
-helm uninstall --namespace "$K8S_NAMESPACE" "$HELM_CHART_NAME" 1>/dev/null || exit 1
-echo "ok."
 echo -n "test-helm.sh] removing $AWS_SERVICE crds installed by Helm ... "
 kubectl delete -f "$HELM_DIR/crds" 1>/dev/null || exit 1
 echo "ok."
+echo -n "test-helm.sh] uninstalling the Helm Chart $HELM_CHART_NAME in namespace $K8S_NAMESPACE ... "
```

Comment:
Can you please explain what is this race condition? I have not seen this test ever fall into any race condition.

In the original code, once helm uninstall finishes, the controller pod is no longer there. And I am having hard time grasping why any controller error will prevent CRD deletion when there is no controller remaining.

Reply:
The race condition occurs when a test errors and resources are not deleted. The controller is not uninstalled when the test fails and there are resources left. Maybe this is just a controller test problem and it should try harder to clean up everything even when a test fails. But uninstalling the helm chart first causes any resource left behind to be orphaned because the finalizers can't complete. This PR moves the CRD deletion before the controller deletion which cleans up any resources left by tests before it brings down the controller. 

To Reproduce:

1) Change a test to fail in a controller e2e test suite (ec2-controller preferably)
2) `make kind-test`
3) Re-run with the same cluster `TMP_DIR.... make kind-test` You should see CRD delete hang because instances of the CRD exist and can't complete their finalizer.

Reply:
I thought about this and I do not think this will really solve the deleting crd problem. The problem will still persist because helm install will not cleanup any custom resources causing the crd deletion to hang.

>"Why are we deleting the crds in first place"

This was added only for local testing. Since we want to e2e-test the controller with the crds inside config/ directory, I delete the crds installed by helm test and then crds from config/crd get installed before e2e tests run.
This all does not matter for Prowjobs because these tests are executed in isolation with each other on prow in different kind clusters

Interestingly, if we do not delete crds in helm tests they will be updated when config/crd is installed using kustomize.

I will raise the PR for that and it will avoid this specific error when tests fail and developer reuses the KinD cluster to execute both helm and e2e tests together.(So specific)